docs/docs/classes.md

```diff
@@ -659,33 +658,54 @@ class AnnotatedClass {
 }
 ```
 
-Annotations are accessed via the `.annotations` property available on all classes. If annotations
-are preset a dictionary is returned, otherwise the `.annotations` property is `nil`.
+```cs
+class ClassWithMethodAnnotation {
+    init() {}
+
+    @MethodAnnotation
+    method() {}
+}
+```
+
+Annotations are access via the `.classAnnotations` and `.methodAnnotations` proprties available on all classes.
+
+For class annotations, the returned data structure returned is a dictionary with keys set to the names of the annotations and their values if present. If no value is provided to the annotation, the value associated with the key is set to `nil`. 
+
+For mathod annotations, the returned data structure is also a dictionary, however the keys are the method names and the values are also dictionaries containing the annotation name and associated values. If no value is provided to the annotation, the value associated with the key is set to `nil`. 
 
 ```cs
-print(AnnotatedClass.annotations); // {"Annotation": nil}
+print(AnnotatedClass.classAnnotations); // {"Annotation": nil}
+```
+
+```cs
+print(ClassWithMethodAnnotation.methodAnnotations); // {"method": {"MethodAnnotation": nil}}
 ```
 
 Annotations can also be supplied a value, however, the value must be of type: nil, boolean, number or string.
 
-```
+```cs
 @Annotation("Some extra value!")
 class AnnotatedClass {
 
 }
 
-print(AnnotatedClass.annotations); // {"Annotation": "Some extra value!"}
+print(AnnotatedClass.classAnnotations); // {"Annotation": "Some extra value!"}
 ```
 
-Multiple annotations can be supplied to classes.
+Method annotations are available on all class methods except initializers.
```

Comment:
Most the thought of class initialization consistency. I'm happy to update it to allow for `init` to have annotations as well.

Reply:
Yeah I think allowing it on constructors could be useful, initial thoughts being it may be useful in the future (potentially?) if we were to create some sort of service container and enable some sort of clever dependency injection or something like that

Reply:
Yeah I think allowing it on constructors could be useful, initial thoughts being it may be useful in the future (potentially?) if we were to create some sort of service container and enable some sort of clever dependency injection or something like that

Reply:
No worries, sorry it took a bit longer! Thanks for the effort on this too

Reply:
Thanks for the review. I think we're all set now.